packages/framework-types/src/concepts/role.ts

```diff
@@ -14,6 +17,12 @@ export interface AuthMetadata {
   readonly skipConfirmation?: boolean
 }
 
+export type BeforeFunction = (
+  filter: FilterFor<Class<ReadModelInterface>>,
+  currentUser?: UserEnvelope
+) => FilterFor<Class<ReadModelInterface>>
```

Comment:
For me, this type makes more sense to be defined in the `./read-model` file, as it has very little to do with roles.  What do you think?

Reply:
Agree! :D